modular/gater/object_handler.go

```diff
@@ -497,13 +476,55 @@ func (g *GateModular) getObjectHandler(w http.ResponseWriter, r *http.Request) {
 
 	} // else anonymous users can get public object.
 
+	// do the actual download
+	err = g.downloadObject(w, reqCtx)
+	if err != nil {
+		return
+	}
+}
+
+func (g *GateModular) downloadObject(w http.ResponseWriter, reqCtx *RequestContext) error {
```

Comment:
add comments for this fucntion .  And can we change the defination of function into  downloadObject(ctx context.Context , w http.ResponseWriter) ? context arg is usually put in the first place

Reply:
comments are added. 
but context arg is not necessary as the this method will use the context of reqCtx :

```
reqCtx.Context()
```

Reply:
we can just pass reqCtx.Context() to downloadObject(ctx context.Context , w http.ResponseWrite), this may make it more general interface?